src/windows/FileTransferProxy.js

```diff
@@ -89,19 +179,141 @@ exec(win, fail, 'FileTransfer', 'upload',
             return;
         }
 
-        if (filePath.substr(0, 8) === "file:///") {
-            filePath = appData.localFolder.path + filePath.substr(8).split("/").join("\\");
-        } else if (filePath.indexOf('ms-appdata:///') === 0) {
-            // Handle 'ms-appdata' scheme
-            filePath = filePath.replace('ms-appdata:///local', appData.localFolder.path)
-                               .replace('ms-appdata:///temp', appData.temporaryFolder.path);
-        } else if (filePath.indexOf('cdvfile://') === 0) {
-            filePath = filePath.replace('cdvfile://localhost/persistent', appData.localFolder.path)
-                               .replace('cdvfile://localhost/temporary', appData.temporaryFolder.path);
+        if (filePath.indexOf("data:") === 0 && filePath.indexOf("base64") !== -1) {
+            // First a DataWriter object is created, backed by an in-memory stream where 
+            // the data will be stored.
+            var writer = Windows.Storage.Streams.DataWriter(new Windows.Storage.Streams.InMemoryRandomAccessStream());
+            writer.unicodeEncoding = Windows.Storage.Streams.UnicodeEncoding.utf8;
+            writer.byteOrder = Windows.Storage.Streams.ByteOrder.littleEndian;
+
+            var commaIndex = filePath.indexOf(",");
+            if (commaIndex === -1) {
+                errorCallback(new FTErr(FTErr.INVALID_URL_ERR, fileName, server, null, null, "No comma in data: URI"));
+                return;
+            }
+
+            // Create internal download operation object
+            fileTransferOps[uploadId] = new FileTransferOperation(FileTransferOperation.PENDING, null);
+
+            var fileDataString = filePath.substr(commaIndex + 1);
+
+            function stringToByteArray(str) {
+                var byteCharacters = atob(str);
+                var byteNumbers = new Array(byteCharacters.length);
+                for (var i = 0; i < byteCharacters.length; i++) {
+                    byteNumbers[i] = byteCharacters.charCodeAt(i);
+                }
+                return new Uint8Array(byteNumbers);
+            };
+
+            // setting request headers for uploader
+            var uploader = new Windows.Networking.BackgroundTransfer.BackgroundUploader();
+            uploader.method = httpMethod;
+            for (var header in headers) {
+                if (headers.hasOwnProperty(header)) {
+                    uploader.setRequestHeader(header, headers[header]);
+                }
+            }
+
+            if (isMultipart) {
+                // adding params supplied to request payload
+                var multipartParams = '';
+                for (var key in params) {
+                    if (params.hasOwnProperty(key)) {
+                        multipartParams += LINE_START + BOUNDARY + LINE_END;
+                        multipartParams += "Content-Disposition: form-data; name=\"" + key + "\"";
```

Comment:
None I've found for Windows and `createUploadFromStreamAsync`. 

I think the current approach works as it is consistent with [Android](https://github.com/apache/cordova-plugin-file-transfer/blob/a9470ff1cc313be542f625f857d33fb14a520d6f/src/android/FileTransfer.java#L385-L411) and [iOS code](https://github.com/apache/cordova-plugin-file-transfer/blob/a9470ff1cc313be542f625f857d33fb14a520d6f/src/ios/CDVFileTransfer.m#L197-L225) for multipart forms and allows a fine tuning.

Alternative could be saving to a temporary file and using `BackgroundTransferContentPart` but it will be less efficient.